openforcefield/tests/test_parameters.py

```diff
@@ -384,6 +384,21 @@ def test_add_delete_cosmetic_attributes(self):
         assert not(bh.attribute_is_cosmetic('pilot'))
 
 
+    def test_get_parameter(self):
+        """Test that ParameterHandler.get_parameter can lookup function
+        """
+        from simtk import unit
+        bh = BondHandler(skip_version_check=True)
+        bh.add_parameter({'smirks': '[*:1]-[*:2]',
+                          'length': 1*unit.angstrom,
+                          'k': 10*unit.kilocalorie_per_mole/unit.angstrom**2})
+
+        params = bh.get_parameter({'smirks': '[*:1]-[*:2]'})
+
+        assert params[0].length == unit.Quantity(1.0, unit.angstrom)
+        assert params[0].k == unit.Quantity(10.0, unit.kilocalorie_per_mole/unit.angstrom**2)
+
```

Comment:
Could you add checks to ensure that:
* a query with no matches returns an empty list
* searching for a nonexistent attr does not raise an exception
* this does indeed behave as an OR query (so, add a statement that would fail an AND)
* this works on optional and cosmetic attributes (like `id` and `blah` respectively)